typescript/web/src/components/core/combo-box/combo-box.tsx

```diff
@@ -0,0 +1,223 @@
+    </Flex>
+  );
+};
+
+const ValuesList = <
+  TValue extends ComboBoxItem<ObjectKey> = Record<ObjectKey, string>,
+  TCompareKey extends keyof TValue = ObjectKey
+>() => {
+  const { items, getMenuProps } = useComboBox<TValue, TCompareKey>();
+  return (
+    <Flex {...getMenuProps()} flexGrow={1} direction="column" overflowY="auto">
+      {items.map((item, index) => (
+        <ValueItem key={item.id} value={item} index={index} />
+      ))}
+    </Flex>
+  );
+};
+
+const SearchableList = () => {
+  const { getComboboxProps } = useComboBox();
+  return (
+    <Flex
+      {...getComboboxProps()}
+      flexGrow={1}
+      direction="column"
+      align="stretch"
+    >
+      <Search />
+      <ValuesList />
+    </Flex>
+  );
+};
+
+const ListPopoverContent = () => (
+  <PopoverContent cursor="default" pointerEvents="initial">
+    <PopoverBody p={0} borderStyle="none" borderWidth={0}>
+      <SearchableList />
+    </PopoverBody>
+  </PopoverContent>
+);
+
+type ListPopoverProps = { trigger: ReactNode };
+
+const ListPopover = ({ trigger }: ListPopoverProps) => {
+  const { isOpen, closeMenu, openMenu } = useComboBox();
+  return (
+    <Popover
+      isOpen={isOpen}
+      onClose={closeMenu}
+      onOpen={openMenu}
+      trigger="click"
+      placement="bottom-start"
+    >
+      <PopoverTrigger>{trigger}</PopoverTrigger>
+      <ListPopoverContent />
+    </Popover>
+  );
+};
+
+const useMainButtonProps = (
+  props: ButtonProps,
+  chakraRef: ForwardedRef<HTMLButtonElement>
+): ButtonProps => {
+  const { getToggleButtonProps } = useComboBox();
+  const { ref: downshiftRef, ...downshiftProps } = getToggleButtonProps();
+  return {
+    ref: useMergeRefs(chakraRef, downshiftRef),
+    "aria-label": "Select value",
+    rightIcon: "selector",
+    minW: 100,
+    variant: "outline",
+    ...downshiftProps,
+    ...props,
+  };
+};
+
+const MainButtonItem = () => {
+  const { item: Item, selectedItem } = useComboBox();
+  return isNil(selectedItem) ? <Spacer /> : <Item {...selectedItem} />;
+};
+
+const MainButton = forwardRef<ButtonProps, "button">((props, chakraRef) => (
+  <Tooltip label="Click to select a value" placement="bottom" openDelay={500}>
+    <Button {...useMainButtonProps(props, chakraRef)}>
+      <MainButtonItem />
+    </Button>
+  </Tooltip>
+));
+
+export type ComboBoxProps<
+  TValue extends ComboBoxItem<ObjectKey> = Record<ObjectKey, string>,
+  TCompareKey extends keyof TValue = ObjectKey
+> = ComboBoxProviderProps<TValue, TCompareKey> &
+  Pick<ButtonProps, "bg" | "borderWidth" | "borderStyle" | "disabled"> & {
+    "data-testid"?: string;
+  };
+
+export const ComboBox = <
+  TValue extends ComboBoxItem<ObjectKey> = Record<ObjectKey, string>,
+  TCompareKey extends keyof TValue = ObjectKey
+>({
+  bg,
+  borderStyle,
+  borderWidth,
+  disabled,
+  "data-testid": testId,
+  ...props
+}: ComboBoxProps<TValue, TCompareKey>) => (
+  <ComboBoxProvider {...props}>
+    <ListPopover
```

Comment:
It's a global defect with all the toolbars of the labeling view 😿 I have clean solutions but it requires heavy changes which someone else could have managed in another PR if correctly groomed 😉 
To workaround this, tooltip delay in this view has been set to 500ms IIRC but I agree that it's dirty and still buggy.
Let's discuss it in the next DSM if you agree so that we can find a better solution 🚀